public/app/features/integrated-alerting/components/Table/Table.types.ts

```diff
@@ -1,8 +1,37 @@
-import { Column } from 'react-table';
+import { Column, TableInstance, TableState, Row, TableOptions } from 'react-table';
 
 export interface TableProps {
   data: object[];
   columns: Column[];
   pendingRequest?: boolean;
   emptyMessage?: string;
+  showPagination?: boolean;
+  totalItems: number;
+  totalPages?: number;
+  tableHash?: string;
+  pageSize: number;
+  pageIndex: number;
+  onPaginationChanged: (pageSize: number, pageIndex: number) => void;
+}
+
+interface ExtendedTableState extends TableState {
```

Comment:
IMO `PaginatedTableState` (or something along those lines) would be a better name for this interface. The term `Extended` is too generic.

Reply:
Seems fair. This interface actually should come from DefinitelyTyped, but it's still outdated at this point, maybe will make a PR there.